### PR TITLE
Extract logging code to the separate file/header.

### PR DESCRIPTION
**Pull request by ni4**
This PR is based on PR #1416 by @dkg and additionally extracts all logging code to the separate code (this would be needed in later PR, which gets rid of lib includes for `rnpcfg.cpp` and `tui.cpp`).

See this comment for more details: https://github.com/rnpgp/rnp/pull/1416#issue-572227080

### CODE REVIEW

**Review by ronaldtse — APPROVED** [review]
LGTM thanks @ni4 ! There's a remaining problem with fuzzing though.

File: src/lib/logging.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ */
+
+#include "string.h"
+#include "logging.h"
+
+/* -1 -- not initialized
+    0 -- logging is off
+    1 -- logging is on
+*/
+int8_t _rnp_log_switch =
```
Comment (antonsviridenko) [inline src/lib/logging.cpp:34]: I guess this variable should not be referenced outside this file, so it makes sense to declare it as `static`

Comment (ni4) [reply src/lib/logging.cpp]: @antonsviridenko Thanks, fixed.

**Comment by ni4** [comment]
@antonsviridenko Any further comments?

**Review by antonsviridenko — APPROVED** [review]

**Pull request merged**